### PR TITLE
[User Model] [Fix] Notification permission change event type to boolean

### PR DESCRIPTION
**Pull request by jkasten2**
# Description
## 1 Line Summary
Correct `OneSignal.Notification.addEventListener("permissionChange", function....)` to provide a `boolean` instead of a `string`.

## Details
# Validation
## Tests
* Tested on Chrome on macOS, ensuring the event fires with a boolean
* Added new automated tests
### Checklist
   - [X] All the automated tests pass or I explained why that is not possible
   - [X] I have personally tested this on my machine or explained why that is not possible
   - [X] I have included test coverage for these changes or explained why they are not needed

**Programming Checklist**
Interfaces:
   - [X] Don't use default export
   - [X] New interfaces are in model files

Functions:
   - [X] Don't use default export
   - [X] All function signatures have return types
   - [X] Helpers should not access any data but rather be given the data to operate on.

Typescript:
   - [X] No Typescript warnings
   - [X] Avoid silencing null/undefined warnings with the exclamation point

Other:
   - [X] Iteration: refrain from using `elem of array` syntax. Prefer `forEach` or use `map`
   - [X] Avoid using global OneSignal accessor for `context` if possible. Instead, we can pass it to function/constructor so that we don't call `OneSignal.context`

### Checklist
   - [X] I have included screenshots/recordings of the intended results or explained why they are not needed

---

## Related Tickets

---

- - -
This change is [Reviewable](https://reviewable.io/reviews/OneSignal/OneSignal-Website-SDK/1075)

### CODE REVIEW

File: __test__/support/managers/PermissionManager.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+export class PermissionManager {
+    public static async mockNotificationPermissionChange(
+        test: jest.It,
+        nativePermission: NotificationPermission,
+      ): Promise<void> {
+        test.stub(RealPermissionManager.prototype, 'getPermissionStatus', nativePermission);
+        // mimick native permission change
+        await PermissionUtils.triggerNotificationPermissionChanged();
+      }
+}
```
Comment (rgomezp) [inline __test__/support/managers/PermissionManager.ts:13]: newline

File: src/shared/utils/PermissionUtils.ts [hunk 2]
```diff
@@ -35,5 +35,14 @@ export class PermissionUtils {
 
     await Database.put('Options', { key: 'notificationPermission', value: newPermission });
     OneSignalEvent.trigger(OneSignal.EVENTS.NOTIFICATION_PERMISSION_CHANGED_AS_STRING, newPermission);
+    
+    const newPermissionBoolean = newPermission === 'granted';
+    const previousPermissionBoolean = previousPermission === 'granted';
+
+    const shouldBeUpdatedBoolean = newPermissionBoolean !== previousPermissionBoolean || updateIfIdentical;
```
Comment (rgomezp) [inline src/shared/utils/PermissionUtils.ts:42]: can shorten to `shouldBeUpdated`. Also maybe add a comment to the function explaining what the argument `updateIfIdentical` does for quick reference.

**Review by rgomezp — CHANGES_REQUESTED** [review]
Overall looks great. A couple nits

File: __test__/support/managers/PermissionManager.ts (hunk 1 above)
Comment (jkasten2) [reply __test__/support/managers/PermissionManager.ts]: Fixed

File: src/shared/utils/PermissionUtils.ts (hunk 2 above)
Comment (jkasten2) [reply src/shared/utils/PermissionUtils.ts]: Fixed the naming on all these and split out the boolean even into it' own function to make things more clear.

**Review by rgomezp — APPROVED** [review]

**Pull request merged**